osu.Game.Rulesets.Mania/Mods/ManiaModFadeIn.cs

```diff
@@ -19,5 +29,86 @@ public class ManiaModFadeIn : Mod
         public override double ScoreMultiplier => 1;
         public override bool Ranked => true;
         public override Type[] IncompatibleMods => new[] { typeof(ModFlashlight<ManiaHitObject>) };
+
+        private const float lanecover_size_filled = 0.5f;
+        private const float lanecover_size_gradient = 0.25f;
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<ManiaHitObject> drawableRuleset)
+        {
+            ManiaPlayfield maniaPlayfield = (ManiaPlayfield)drawableRuleset.Playfield;
+
+            foreach (Column column in maniaPlayfield.Stages.SelectMany(stage => stage.Columns))
+            {
+                column.hitObjectArea.ChildrenOfType<BufferedContainer>().First().Add(new LaneCover(false)
```

Comment:
I am well aware that this particular line is awful, I just couldn't figure out how else to do it. Please advise

Reply:
* should be a `Single()` if anything to begin with
* should not use `ChildrenOfType()` but a named field to which the buffered container is assigned

that said, I would explore options as to whether the buffered container can be slotted in solely for the fade-in mod, instead of being added globally as done here.